app/src/main/java/com/okta/appauth/android/example/TestLoginActivity.java

```diff
@@ -0,0 +1,172 @@
+public class TestLoginActivity extends AppCompatActivity {
+
+    private static final String TAG = "TestLoginActivity";
+
+
+    private View mLoginLayout;
+    private ProgressBar mAuthProgress;
+    private Button mButton;
+    private View mAuthorizedInfo;
+    private TextView mIdToken;
+    private TextView mAccesToken;
+    private Okta okta;
+    private ExecutorService loginExecutor = Executors.newSingleThreadExecutor();
+    private Future loginTask;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        Log.d(TAG, "onCreate");
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.test_activity_login);
+        mLoginLayout = findViewById(R.id.loading_container);
+        mButton = findViewById(R.id.start_button);
+        mIdToken = findViewById(R.id.id_token_info);
+        mAccesToken = findViewById(R.id.access_token_info);
+        mAuthorizedInfo = findViewById(R.id.auth_container);
+        mAuthProgress = findViewById(R.id.progress_bar);
+
+        mButton.setOnClickListener(v -> performBrowserLogin());
+        initOkta();
+
+        //In case our activity died during authentication
+        //and have been recreated
+        if (okta.provideOktaState().hasPendingAthentication()) {
+            performBrowserLogin();
+        }
+    }
+
+    @Override
+    protected void onResume() {
+        super.onResume();
+        showAuthInfo();
+    }
+
+    private void performBrowserLogin() {
+        loginTask = loginExecutor.submit(() -> {
+            startLogin();
+            AuthorizationResult result = okta.authenticateWithBrowser(this, null);
+            stopLogin();
+            if (result.isSuccess()) {
+                showAuthInfo();
+            } else {
+                showSnackbar("Login error " + result.getError().getMessage());
+            }
+        });
+    }
+
+    private void showAuthInfo() {
+        mButton.post(() -> {
+            if (okta.provideOktaState().isAuthenticatedd()) {
+                mAccesToken.setText(okta.provideOktaState().getTokens().getAccessToken());
+                mIdToken.setText(okta.provideOktaState().getTokens().getIdToken());
+                mAuthorizedInfo.setVisibility(View.VISIBLE);
+                mLoginLayout.setVisibility(View.GONE);
+            }
+        });
+    }
+
+    private void initOkta() {
+        OktaConfig config = new OktaConfig.Builder()
+                .withClientId("0oahnzhsegzYjqETc0h7")
+                .withRedirectUri("com.lohika.android.test:/callback")
+                .withEndSessionRedirectUri("com.lohika.android.test:/logout")
+                .withScopes("openid", "profile", "offline_access")
+                .withIssuerUri("https://lohika-um.oktapreview.com/oauth2/default")
+                .build();
+
+        okta = new OktaBuilder()
+                .withOktaFactory(new PlainOktaFactory())
+                .withConfig(config)
+                .withContext(this.getApplicationContext())
```

Comment:
Do we still need getAppContext? Since its not being used anymore?

Reply:
Good point will remove it.